File: rapid-router/src/main/java/com/codeforlife/rapidrouter/MainActivity.java
package com.codeforlife.rapidrouter;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.google.blockly.android.AbstractBlocklyActivity;
import com.google.blockly.android.codegen.CodeGenerationRequest;
import com.google.blockly.android.codegen.LoggingCodeGeneratorCallback;
import com.google.blockly.model.DefaultBlocks;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MainActivity extends AbstractBlocklyActivity {
    private final CodeGenerationRequest.CodeGeneratorCallback generatorCallback =
            new LoggingCodeGeneratorCallback(this, "LoggingTag");

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected View onCreateContentView(int parentId) {
        return getLayoutInflater().inflate(R.layout.layout, null);
    }

    @NonNull
    @Override
    protected String getToolboxContentsXmlPath() {
        return "toolbox/toolbox37.xml";
    }

    @NonNull
    @Override
    protected List<String> getBlockDefinitionsJsonPaths() {
        final ArrayList<String> assetPaths = new ArrayList<>(DefaultBlocks.getAllBlockDefinitions());
        assetPaths.add("blocks/CustomBlocks.json");
        return assetPaths;
    }

    @NonNull
    @Override
    protected List<String> getGeneratorsJsPaths() {
        return Collections.emptyList();
    }

    @NonNull
    @Override
    protected CodeGenerationRequest.CodeGeneratorCallback getCodeGenerationCallback() {
        return generatorCallback;
    }
}

File: rapid-router/src/main/java/com/codeforlife/rapidrouter/MainActivity.kt
package com.codeforlife.rapidrouter

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import com.badlogic.gdx.backends.android.AndroidFragmentApplication
import com.codeforlife.rapidrouter.fragments.BlocklyFragment
import com.codeforlife.rapidrouter.fragments.GameFragment
import com.codeforlife.rapidrouter.utils.LevelReader


class MainActivity : AppCompatActivity(), AndroidFragmentApplication.Callbacks {

    override fun exit() {

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout)

        val pager = findViewById<ViewPager>(R.id.view_pager)
        pager.adapter = MainViewPagerAdapter(supportFragmentManager)

        supportActionBar?.elevation = 0.0f

        val loadLevel = LevelReader.loadLevel(21, assets)
    }
}

class MainViewPagerAdapter(manager: FragmentManager) : FragmentStatePagerAdapter(manager) {

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            BlocklyFragment()
        } else GameFragment()
    }

    override fun getCount() = 2

    override fun getPageTitle(position: Int) = if (position == 0) "Blockly Stage" else "Game View"

}